zqd/api/api.go

```diff
@@ -45,6 +45,19 @@ type SearchRequest struct {
 	Dir   int             `json:"dir" validate:"required"`
 }
 
+type WorkerRequest struct {
+	SearchRequest
+	Chunks []ChunkEntry `json:"chunks"`
+}
+
+type ChunkEntry struct {
+	Id           string  `json:"ksuid" validate:"required"`
+	First        nano.Ts `json:"first" validate:"required"`
+	Last         nano.Ts `json:"last" validate:"required"`
+	DataFileKind string  `json:"filekind" validate:"required"`
+	RecordCount  int     `json:"recordcount" validate:"required"`
```

Comment:
```suggestion
	Id           string  `json:"id" validate:"required"`
	First        nano.Ts `json:"first" validate:"required"`
	Last         nano.Ts `json:"last" validate:"required"`
	FileKind string  `json:"data_file_kind" validate:"required"`
	RecordCount  int     `json:"record_count" validate:"required"`
```
For JSON field names we use the snake-case version of the Go field name.